service/src/main/java/tds/assessment/repositories/impl/AssessmentQueryRepositoryImpl.java

```diff
@@ -147,6 +149,51 @@ public AssessmentQueryRepositoryImpl(final NamedParameterJdbcTemplate jdbcTempla
     }
 
     @Override
+    public List<AssessmentInfo> findAssessmentInfoByKeys(final String clientName, final String... assessmentKeys) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("keys", Arrays.asList(assessmentKeys))
+            .addValue("clientName", clientName);
+
+        final String SQL =
+            "SELECT \n" +
+                "   a._key AS assessmentKey, \n" +
+                "   a.testid AS assessmentId, \n" +
+                "   tp.subjectname AS subject, \n" +
+                "   tp.label AS assessmentLabel, \n" +
+                "   GROUP_CONCAT(DISTINCT tool.code) languages, \n" +
+                "   GROUP_CONCAT(DISTINCT g.grade) grades, \n" +
+                "   tp.maxopportunities AS maxAttempts \n" +
```

Comment:
The GROUP_CONCAT function is not something I had used before - it groups multiple rows of values into one comma delimited list (for example, an assessment with grades "6" and "7" would end up with the grades column as "6,7". We then parse this in the RowMapper with a simple .split(","). I thought this might be a bit of a simpler approach then some of the other solutions we are currently using, but feel free to point out any flaws with this.